File: examples/complex/process-bar/python/_cott_impl/foo/bar/validate_payload.py
from cott_runtime import Err, Ok, Result
from foo.bar_types import BarError, BarError_InvalidPayload, InputPayload


def validate_payload(data: InputPayload) -> Result[InputPayload, BarError]:
    if len(data.data) == 0:
        return Err(error=BarError_InvalidPayload(reason="payload data must not be empty"))
    return Ok(value=data)

File: examples/kotlin/complex/process-bar/kotlin/cott_impl/foo/bar/validate_payload.kt
package cott_impl.foo.bar

internal fun validate_payload(`data`: foo.bar.InputPayload): cott_runtime.CottResult<foo.bar.InputPayload, foo.bar.BarError> =
    if (`data`.data.isEmpty()) {
        cott_runtime.Err(foo.bar.BarError.InvalidPayload(reason = "Payload bytes must not be empty"))
    } else {
        cott_runtime.Ok(`data`)
    }
